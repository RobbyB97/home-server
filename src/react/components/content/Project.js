import React from 'react'


export class Project extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            active: false
        }
    }

    toggle = () => {
        this.setState({active: !this.state.active})
    }

    toggleOn = () => {
        this.setState({active: true})
    }

    toggleOff = () => {
        this.setState({active: false})
    }

    render() {
        return (
            <div
                className={this.state.active ? 
                    "project active" : 
                    "project"}
                
                onMouseEnter={this.toggleOn}
                onMouseLeave={this.toggleOff}
            >
                {this.state.active ?
                    <div className="project__active">
                        
                    </div>
                :
                    <div className="project__inactive">
                        {this.props.logo &&
                            <img 
                                className="project__logo"
                                src={this.props.logo}
                            />
                        }

                        <p className="project__name">
                            {this.props.name}
                        </p>
                    </div>
                }
            </div>
        )
    }
}


export default Project