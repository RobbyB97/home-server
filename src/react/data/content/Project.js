export const project_ytIframe = {
    name: 'yt-iframe',
    languages: 'Python',
    link: 'https://pypi.org/project/yt-iframe/',
    linkText: 'View on PyPI!',
    description: 'yt-iframe is an HTML file generator written in Python. All you need to do is add a link to a youtube channel and an HTML file will be generated that cleanly displays all the youtube videos of a channel.'
}

export const project_thisWebsite = {
    name: 'This website!',
    languages: 'React, Node',
    link: 'https://github.com/RobbyB97/home-server',
    description: 'This website is a React/Node project originally hosted on my home server, but recently migrated to Heroku. (They should really add warning labels to servers about electricity bills)'
}

export const project_webDevBootcamp = {
    name: 'Web dev bootcamp',
    languages: 'React (Router, Redux, Native), Node, Django, Electron',
    link: 'https://github.com/RobbyB97/web-dev-bootcamp',
    description: 'This repository hosts the codebase for personal "dummy" projects I create while learning the basics of different web technologies.'
}