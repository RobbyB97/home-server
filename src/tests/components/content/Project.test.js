import React from 'react'
import {shallow} from 'enzyme'

import Project from '../../../react/components/content/Project'


test('Render Project', () => {
    const wrapper = shallow(<Project />)
    
    expect(wrapper).toMatchSnapshot()
})