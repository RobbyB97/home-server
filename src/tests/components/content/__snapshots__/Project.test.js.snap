// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Project 1`] = `
<div
  className="project"
/>
`;
